devicetypes/axis/axis-gear-v5.1.src/axis-gear-v5.1.groovy

```diff
@@ -0,0 +1,420 @@
+		}
+        
+        standardTile("home", "device.level", width: 2, height: 2, decoration: "flat") {
+            state "default", label: "Preset", action:"presetPosition", icon:"st.Home.home2"
+        }
+        
+        preferences {
+            input "preset", "number", title: "Preset percentage (1-100) [Default - 50%]", defaultValue: 50, required: false, displayDuringSetup: true, range:"(1..100)"
+        }
+        
+        main(["main"])
+        details(["windowShade", "mediumSlider", "contPause", "home", "version", "battery", "refresh"])
+
+        //details('shade', 'levelSliderControl')
+        //details('levelSliderControl', 'otherTile', 'anotherTile') //adjustment and order of tiles
+	}
+
+
+}
+
+//Declare Clusters
+private getCLUSTER_BASIC() {0x0000}
+private getBASIC_ATTR_SWBUILDID() {0x4000}
+
+private getCLUSTER_POWER() {0x0001}
+private getPOWER_ATTR_BATTERY() {0x0021}
+
+private getCLUSTER_ONOFF() {0x0006}
+private getONOFF_ATTR_ONOFFSTATE() {0x0000}
+
+private getCLUSTER_LEVEL() {0x0008}
+private getLEVEL_ATTR_LEVEL() {0x0000}
+private getLEVEL_CMD_STOP() {0x03}
+
+private getCLUSTER_WINDOWCOVERING() {0x0102}
+private getWINDOWCOVERING_ATTR_LIFTPERCENTAGE() {0x0008}
+private getWINDOWCOVERING_CMD_OPEN() {0x00}
+private getWINDOWCOVERING_CMD_CLOSE() {0x01}
+private getWINDOWCOVERING_CMD_STOP() {0x02}
+private getWINDOWCOVERING_CMD_GOTOLIFTPERCENTAGE() {0x05}
+
+//String currentVersionString = "";
+
+//Custom command to increment blind position by 25 %
+def ShadesUp(){
+	def shadeValue = device.latestValue("level") as Integer ?: 0 
+    
+    if (shadeValue < 100){
+      	shadeValue = Math.min(25 * (Math.round(shadeValue / 25) + 1), 100) as Integer
+    }else { 
+    	shadeValue = 100
+	}
+    //sendEvent(name:"level", value:shadeValue, displayed:true)
+    setLevel(shadeValue)
+    //sendEvent(name: "windowShade", value: "opening")
+}
+
+//Custom command to decrement blind position by 25 %
+def ShadesDown(){
+	def shadeValue = device.latestValue("level") as Integer ?: 0 
+    
+    if (shadeValue > 0){
+      	shadeValue = Math.max(25 * (Math.round(shadeValue / 25) - 1), 0) as Integer
+    }else { 
+    	shadeValue = 0
+	}
+    //sendEvent(name:"level", value:shadeValue, displayed:true)
+    setLevel(shadeValue)
+    //sendEvent(name: "windowShade", value: "closing")
+}
+
+def stop() {
+    log.info "stop()"
+    def shadeState = device.latestValue("windowShade")
+    if (shadeState == "opening" || shadeState == "closing") {
+        if (state.currentVersion >= 1093){
+            sendEvent(name: "windowShade", value: "stopping")
+
+            return zigbee.command(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_CMD_STOP)         
+        }
+        else {
+            sendEvent(name: "windowShade", value: "stoppingNS")
+            return zigbee.readAttribute(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL, [delay:5000])        
+        }
+    }
+    else {
+    	if (state.currentVersion >= 1093){
```

Comment:
this repeated use of `1093` should be replaced with a function `getMIN_WINDOW_COVERING_VERSION` like you've done for the clusters.